endpoints/openrtb2/auction.go

```diff
@@ -719,8 +726,21 @@ func (deps *endpointDeps) validateRequest(req *openrtb_ext.RequestWrapper, isAmp
 		return []error{err}
 	}
 
-	if (req.Site == nil && req.App == nil) || (req.Site != nil && req.App != nil) {
-		return append(errL, errors.New("request.site or request.app must be defined, but not both."))
+	// Prep for mutual exclusion check
+	invTypeNumMatches := 0
+	if req.Site != nil {
+		invTypeNumMatches++
+	}
+	if req.App != nil {
+		invTypeNumMatches++
+	}
+	if req.DOOH != nil {
+		invTypeNumMatches++
+	}
+	if invTypeNumMatches == 0 {
+		return append(errL, errors.New("One of request.site or request.app or request.dooh must be defined"))
+	} else if invTypeNumMatches >= 2 {
+		return append(errL, errors.New("No more than one of request.site or request.app or request.dooh can be defined"))
```

Comment:
+1

I think this code is used in a few other places. Here's one recommendation for envisioning separate methods:

```
if ortb.HasNoInventoryType(req.Site, req.App, req.DOOH) {
  return append(errL, errors.New("one of request.site, request.app, or request.dooh must be defined."))
} else if ortb.HasMoreThanOneInventoryType(req.Site, req.App, req.DOOH) {
  return append(errL, errors.New("only one of request.site, request.app, or request.dooh can be defined."))
}
```

Reply:
Opted to refactor to a single function instead of two as proposed above - I found it more legible and avoided having to do the actual logic check in two functions (hasNo vs HasMoreThanOne)